pkg/addon/common.go

```diff
@@ -182,15 +182,22 @@ func (pa *PolicyAgentAddon) Manifests(
 		return nil, errors.New("the Policy Addon controller is paused due to the policy-addon-pause annotation")
 	}
 
+	var hostingCluster *clusterv1.ManagedCluster
+
 	// Fetch ManagedCluster for hosting cluster in hosted mode
 	hostingClusterName := addon.Annotations[addonapiv1alpha1.HostingClusterNameAnnotationKey]
 	if pa.GetAgentAddonOptions().HostedModeEnabled && hostingClusterName != "" {
-		hostingCluster, err := pa.clusterLister.Get(hostingClusterName)
+		var err error
+
+		hostingCluster, err = pa.clusterLister.Get(hostingClusterName)
 		if err != nil {
 			return nil, err
 		}
+	}
 
-		pa.hostingCluster = hostingCluster
+	helmAgentAddon, ok := pa.AgentAddon.(*addonfactory.HelmAgentAddon)
+	if ok {
+		helmAgentAddon.SetHostingCluster(hostingCluster)
```

Comment:
Shouldn't this be inside the conditional above? Otherwise, it's just setting the `hostingCluster` to `nil`, which doesn't seem necessary.

Reply:
@dhaiducek this is to handle the case where the same addon is in hosted mode on one cluster but another isn't.

Reply:
Isn't setting the hosting cluster more appropriate inside of the `pa.GetAgentAddonOptions().HostedModeEnabled` conditional, though? Otherwise `hostingCluster` is `nil`?